client-core/src/wallet/default_wallet_client.rs

```diff
@@ -183,8 +183,10 @@ where
         m: usize,
         n: usize,
     ) -> Result<ExtendedAddr> {
-        // To verify if the passphrase is correct or not
-        self.transfer_addresses(name, passphrase)?;
+        let wallet_public_keys = self.public_keys(&name, &passphrase)?;
+        if !wallet_public_keys.contains(&self_public_key) {
```

Comment:
This check is already performed [here](https://github.com/crypto-com/chain/blob/master/client-core/src/service/root_hash_service.rs#L53). Remove the redundant check here and we can improve error reporting in underlying function.

Reply:
If I understand correctly, this check is talking about the public key list argument contains the self public key? The check I added is trying to make sure the self key belongs to the wallet.

Reply:
Yeah. I misunderstood.